src/Peripherals/PeripheralInterface.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+class QString;
+
+class PeripheralInterface
+{
+public:
+    virtual bool attemptConnection()=0;
+    virtual void setPortName(QString portname)=0;
```

Comment:
Setting portnames is specific to a serial port, we should have a general arguments function instead. `setParamaters`, takes in a list of strings. For example, the udpPeripheral will need an ip address.